src/Entity/Index/EntityIndexInterface.php

```diff
@@ -0,0 +1,33 @@
+<?php
+
+namespace Drupal\entity\Entity\Index;
+
+use Drupal\Core\Entity\EntityInterface;
+
+interface EntityIndexInterface {
+
+  /**
+   * @param string $key
+   *
+   * @return array
+   */
+  public function get($key);
```

Comment:
In `EntityIndexBase` the key is entity type id followed by the entity id, for example `node:1`, this is pretty useless, but anything that extends this like `UuidIndex` does can set what their own key would be. `UuidIndex` uses the uuid.